airflow/models/dag.py

```diff
@@ -734,15 +759,22 @@ def task_group(self) -> "TaskGroup":
 
     @property
     def filepath(self) -> str:
-        """File location of where the dag object is instantiated"""
-        fn = self.full_filepath.replace(settings.DAGS_FOLDER + '/', '')
-        fn = fn.replace(os.path.dirname(__file__) + '/', '')
+        """:meta private:"""
+        warnings.warn(
+            "filepath is deprecated, use relative_fileloc instead", DeprecationWarning, stacklevel=2
+        )
+        return self.relative_fileloc
+
+    @property
+    def relative_fileloc(self) -> str:
+        """File location of the importable dag 'file' relative to the configured DAGs folder."""
+        fn = self.fileloc.replace(settings.DAGS_FOLDER + '/', '')
```

Comment:
Now strange boundaries may appear when the user has a non-standard DAG folder path e.g. `/dags/`. If the user then creates another `/dag/`  folder in this directory, they will get the wrong path
```
>>> '/dags/tomek/dags/'.replace('/dags/', '')
'tomek'
```
Do you consider to use [os.path.relpath](https://docs.python.org/3/library/os.path.html#os.path.relpath) instead?

Reply:
Good idea. I hadn't yet looked through the implications of this, but this was what we were previously doing, just else where in the codebase.

Reply:
Let’s also fix those then (maybe in a separate PR before this one is merged). Using `relpath` improves Windows compatibility as well.